File: Sorting/bubbleSort.cpp
#include<bits/stdc++.h>
using namespace std;

void bubbleSort(vector<int>& arr) {
    /*
    * Time Complexity: O(N^2)
    * Space Complexity: O(1)
    */
    for (int i = 0; i < (int)arr.size(); i++) {
        for (int j = i + 1; j < (int)arr.size(); j++) {
            if (arr[i] > arr[j])
                swap(arr[i], arr[j]);
        }
    }
}


int main() {
    vector<int> arr = { 8, 5, 2, 10, 7, 4, -1,-6, 0 };

    bubbleSort(arr);

    for (int i = 0; i < (int)arr.size(); i++) {
        cout << arr[i] << ' ';
    }

    cout << '\n';

}

File: Sorting/BubbleSort.py
def bubbleSort(arr):
	'''
	Time Complexity : O(N^2)
	Space Complexity : O(1)
	'''
	for i in range(len(arr)):
		for j in range(i+1,len(arr)):
			if(arr[i]>arr[j]):
				arr[i],arr[j]=arr[j],arr[i]
	return arr

arr=[9,4,3,2,5,1,6,7]
print(bubbleSort(arr))

# Code Link: https://ideone.com/rrKGUA